auditor/src/domain/record.rs

```diff
@@ -181,7 +181,7 @@ pub struct RecordUpdate {
 
 /// A `Record` represents a single accountable unit.
 ///
-/// Records can be sent to and received from Auditor with the [`AuditorClient`](`crate::client::AuditorClient`).
+/// Records can be sent to and received from Auditor with the [`AuditorClient`](`/auditor_client::AuditorClient`).
```

Comment:
This does not produce a correct link. When I run `cargo doc --open` on my machine, this link points to `file:///home/benjamin/Code/AUDITOR/target/doc/auditor/domain/%60/auditor_client::AuditorClient%60`. Instead, it should point to `file:///home/benjamin/Code/AUDITOR/target/doc/auditor_client/struct.AuditorClient.html`. I fear that this then also won't work when pushing to docs.rs.
Unfortunately, I also don't really know how to do proper cross-crate references in the docs ...

Reply:
You are right, the links are broken. This will not work on docs.rs.
It seems like there is no working mechanism to reference parts of a crate that you don't depend on.
We *could* use relative links like this:
```
/// Records can be sent to and received from Auditor with the
/// [`AuditorClient`](../../auditor_client/struct.AuditorClient.html) ...
```
But these need to be maintained manually, which I don't like.

Instead, let's move the tutorial into the `auditor_client` crate as you suggested.
As far as I can see the only remaining broken link would be in `auditor/src/domain/record.rs`. There I would only mention the client crate:
```
/// Records can be sent to and received from Auditor with the
/// [`AuditorClient`](../../auditor_client/index.html) crate.
```
Then at least changes to the client crate will not break this link.